pymodbus/transaction.py

```diff
@@ -436,13 +435,40 @@ def delTransaction(self, tid):
         _logger.debug("Deleting transaction %d" % tid)
         if self.transactions: self.transactions.pop(0)
 
+
+class LatestTransactionManager(FifoTransactionManager):
+    """ Implements a transaction for a manager where the
+    results is the latest transaction only. This can be used where only the
+    latest transaction matters.
+    """
+
+    def getTransaction(self, tid):
+        """ Returns a transaction matching the referenced tid
+
+        If the transaction does not exist, None is returned
+
+        As no tids are stored, retrieve the latest transaction
+        :param tid: The transaction to retrieve: Not used
+        """
+        return self.transactions.pop() if self.transactions else None
+
+    def delTransaction(self, tid):
```

Comment:
Just figured out that this method is not at all used except in tests. I would suggest to use `FIFOTransactionManager` class as before .

Reply:
In my testing it definately was called. Each of the cancelled tasks due to timeout were listed in transactions list, and using FIFO meant that it never recovered from timeout error.